### PR TITLE
Add blank table handling for new returns from heasarc

### PR DESCRIPTION
**Pull request by keflavich**
HEASARC is returning FITS files with blank data and a little header metadata as null results now, so we need to handle them.

WIP: requires a test.

### CODE REVIEW

**Comment by bsipocz** [comment]
@keflavich - can you rebase, that should fix CI.

**Comment by keflavich** [comment]
Now I'm confused.  The test code is explicitly catching the warning that is getting raised, following the [python guidelines](https://docs.python.org/3/library/warnings.html#testing-warnings), but the test code is [still failing on the warning](https://github.com/astropy/astroquery/actions/runs/3671969240/jobs/6207697684#step:5:526).

File: astroquery/heasarc/tests/test_heasarc_remote.py [hunk 1]
```diff
@@ -96,3 +100,19 @@ def test_query_region(self):
...
 
         assert len(table) == 63
+
+    def test_query_region_nohits(self):
+        """
+        Regression test for #2560: HEASARC returns a FITS file as a null result
+        """
+        heasarc = Heasarc()
+
+        with warnings.catch_warnings(record=True) as warn:
```
Comment (bsipocz) [inline astroquery/heasarc/tests/test_heasarc_remote.py:112]: why not use `pytest.warns` here?

**Review by bsipocz — CHANGES_REQUESTED** [review]
This will need a changelog entry. And could you squash the testing iterations a bit?

And I see some remote tests failings

File: astroquery/heasarc/tests/test_heasarc_remote.py (hunk 1 above)
Comment (keflavich) [reply astroquery/heasarc/tests/test_heasarc_remote.py:112]: because I didn't remember it was a thing

**Comment by keflavich** [comment]
it passed, I squashed, has changelog.  Going to bed.  $5 says it'll break and I'll need to do more in the morning.

**Comment by bsipocz** [comment]
> it passed, I squashed, has changelog. Going to bed. $5 says it'll break and I'll need to do more in the morning.

haha, the only thing that breaks atm is that in the squash you left the `[skip ci]` in the commit message, so tests are not running. Otherwise I would think this passes, so I do a review and force the CI to pass to get this in before you wake up 😉

**Review by bsipocz — APPROVED** [review]
This is all green now, and changes look good. Ideally the problem should be reported upstream (no results wrapped in a fitsfile is more than weird), but that's out of scope for the PR.

**Comment by bsipocz** [comment]
Thanks Adam!

**Pull request merged**